onload/cplane/main.go

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: (c) Copyright 2023 Advanced Micro Devices, Inc.
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	cplaneServerPathFilePath       = "/sys/module/onload/parameters/cplane_server_path"
+	cplaneServerParamsFilePath     = "/sys/module/onload/parameters/cplane_server_params"
+	critclFilePath                 = "/usr/bin/crictl"
+	crioPrefix                     = "cri-o://"
+	cplaneServerParamsFormatString = "exec %s /opt/onload/sbin/onload_cp_server -K"
+
+	timeoutTime = 5 * time.Second
+)
+
+var (
+	podName      string
+	podNamespace string
+
+	containerID string
+)
+
+func warnAndWait(message string) {
+	glog.Warningf("%s. Will retry in %v", message, timeoutTime)
+	time.Sleep(timeoutTime)
+}
+
+func main() {
+
+	podName = os.Getenv("POD_NAME")
+	podNamespace = os.Getenv("POD_NAMESPACE")
+
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		glog.Fatalf("Failed to get cluster config (%v)", err)
+	}
+
+	clientSet, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		glog.Fatalf("Failed to create clientset (%v)", err)
+	}
+	pod := &corev1.Pod{}
+
+	// The information about the pod may be out of date when it is received,
+	// so we enter a "while" loop to keep checking about the pod and container
+	// until they report as running.
+	for {
+		pod, err = clientSet.CoreV1().Pods(podNamespace).Get(
```

Comment:
Shall we move the "get container ID(s?) by pod namespace and name" part into a separate function?